awx/ui/test/e2e/tests/test-pagination.js

```diff
@@ -0,0 +1,79 @@
+import {
+    getAdminMachineCredential,
+    getInventorySource,
+    getJobTemplate,
+    getOrganization,
+    getProject,
+} from '../fixtures';
+
+import {
+    AWX_E2E_TIMEOUT_MEDIUM,
+} from '../settings';
+
+const namespace = 'test-pagination';
+
+module.exports = {
+
+    before: (client, done) => {
+        const resources = [
+            getAdminMachineCredential(namespace),
+            getOrganization(namespace),
+            getProject(namespace),
+            getInventorySource(namespace),
+        ];
+
+        for (let i = 0; i < 25; i++) {
+            // Create enough job templates to make at least 2 pages of data
+            resources.push(getJobTemplate(namespace, 'hello_world.yml', `${namespace}-job-template-${i}`, false));
+        }
+
+        Promise.all(resources)
+            .then(() => {
+                done();
+            });
+
+        client
+            .login()
+            .waitForAngular()
+            .resizeWindow(1200, 1000);
+    },
+
+    'Test job template pagination operations': client => {
```

Comment:
We should call this out as pagination with filter. We should also have a general pagination without filter test